src/domain/task-system/services/src/task_agent_impl.rs

```
@@ -149,15 +149,18 @@ impl TaskAgentImpl {
         );
 
         // Prepare task definition (requires transaction)
-        let task_definition = DatabaseTransactionRunner::new(self.catalog.clone())
+        let Ok(task_definition) = DatabaseTransactionRunner::new(self.catalog.clone())
             .transactional_with(
                 |task_definition_planner: Arc<dyn TaskDefinitionPlanner>| async move {
                     task_definition_planner
                         .prepare_task_definition(&task.logical_plan)
                         .await
                 },
             )
-            .await?;
+            .await
+        else {
+            return Ok(TaskOutcome::Failed(TaskError::Empty));
```

Add a trace for the error, it is swallowed completely right now.

---

We already have on the inside level, but yeah, will add here too